src/include/duckdb/parser/constraints/foreign_key_constraint.hpp

```diff
@@ -0,0 +1,37 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/parser/constraints/foreign_key_constraint.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/parser/constraint.hpp"
+#include "duckdb/common/vector.hpp"
+
+namespace duckdb {
+
+class ForeignKeyConstraint : public Constraint {
+public:
+	DUCKDB_API ForeignKeyConstraint(vector<string> pk_columns, vector<string> fk_columns, ForeignKeyInfo info);
+
+	//! The set of main key table's columns
+	vector<string> pk_columns;
+	//! The set of foreign key table's columns
+	vector<string> fk_columns;
+	ForeignKeyInfo info;
```

Comment:
This still contains information that cannot be known at parse time (the column indexes). Could we move this out of the constraint, and use this only in the bound constraint?

Reply:
I have already explained this before. We need to have index information for keys because we cannot lookup entry when binding with respect to deaklock.